src/task/OnceTask.ts

```diff
@@ -0,0 +1,109 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import Logger from '../logger/Logger';
+import LogLevel from '../logger/LogLevel';
+import Task from './Task';
+
+function nextTick(f: () => void): void {
```

Comment:
Should we use AsyncScheduler here since you introduced nextTick there already?

Reply:
Sure.